tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -166,14 +166,20 @@ def general_data_processing(self, X: DataFrame, X_val: DataFrame, X_unlabeled: D
         """ General data processing steps used for all models. """
         X = copy.deepcopy(X)
 
-        # Remove all examples with missing labels from this dataset:
-        missinglabel_inds = list(X[X[self.label].isna()].index)
-        if len(missinglabel_inds) > 0:
-            logger.warning(f"Warning: Ignoring {len(missinglabel_inds)} (out of {len(X)}) training examples for which the label value in column '{self.label}' is missing")
-            X = X.drop(missinglabel_inds, axis=0)
+        if pd.api.types.is_float_dtype(X[self.label]):
+            invalid_labels = ~np.isfinite(X[self.label])
+            if np.any(invalid_labels):
+                first_invalid_label_idx = invalid_labels.argmax()
+                raise ValueError(f"Label column cannot contain non-finite values (NaN, Inf, Ninf). First invalid label at idx: {first_invalid_label_idx}")
 
         holdout_frac_og = holdout_frac
         if X_val is not None and self.label in X_val.columns:
+            if pd.api.types.is_float_dtype(X[self.label]):
+                invalid_tuning_labels = ~np.isfinite(X_val[self.label])
+                if np.any(invalid_tuning_labels):
+                    first_invalid_label_idx = invalid_tuning_labels.argmax()
```

Comment:
would argmax correctly work on np.NINF?

Reply:
Yes, it does work because `np.isfinite()` returns numpy array of binary values where all finite values are `True` and all others (`nan`, `inf`, `ninf`) will be `False`. I negate that array and if `any` of the entries are `True`, I return `argmax()` which will return the index of the first `True` entry, which could be any sort of non-finite value.

I spent a while trying to file the most pythonic way to find the index of the first `True` value in a numpy array of binary values, and `argmax()` was the best I could come up with. If you know of a more elegant way to accomplish this, I would be happy to change it because `argmax()` does seem a bit out of place here.

Reply:
I update the unit test to demonstrate it works for `np.inf` and `np.NINF`